app/models/setting/discovered.rb

```diff
@@ -72,6 +74,10 @@ def self.discovery_hostname_fact_array
     from_array Setting['discovery_hostname']
   end
 
+  def self.discovery_lock?
+    Foreman::Cast.to_bool(Setting['discovery_lock']) and !Setting['discovery_lock_template'].blank? and ProvisioningTemplate.exists?(:name => Setting['discovery_lock_template'])
```

Comment:
Hmm isn't the setting automatically created with boolean type so no casting is necessary? You may created it with incorrect type (e.g. string) so it does not work for you. Try to delete it and let it recreate - it should automatically create a boolean type from the default value I think. If not, keep it as is.